File: two pointers/intersection_of_two_arrays2.java
import java.util.*;
// two pointers and sorting
// tc O(nlogn+mlogm), sc O(n)
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        int p1=0, p2=0;
        ArrayList<Integer> result = new ArrayList<Integer>();
        while(p1<nums1.length && p2<nums2.length){
            if(nums1[p1]<nums2[p2]){
                p1++;
            }
            else if(nums2[p2]<nums1[p1]){
                p2++;
            }
            else{
                result.add(nums1[p1]);
                p1++;
                p2++;
            }
        }
        int[] res= new int[result.size()];
        for(int i=0;i<result.size();i++){
            res[i] = result.get(i);
        }
        return res;
    }
}

File: two pointers/intersection_of_two_arrays2.py
# two pointers and sorting
# tc O(mlogm+nlogn), sc O(n)
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        p1, p2 = 0, 0
        result = []
        while p1<len(nums1) and p2<len(nums2):
            if nums1[p1]>nums2[p2]:
                p2+=1
            elif nums2[p2]>nums1[p1]:
                p1+=1
            else:
                result.append(nums1[p1])
                p1+=1
                p2+=1
        return result